backend/WikiContrib/result/views.py

```diff
@@ -183,22 +199,123 @@ async def get_gerrit_data(url, session, gerrit_resp,full_names):
     await get_full_name(data={"platform":"gerrit","full_names":full_names,"url":url,"gerrit_response":data})
 
 
-def format_data(pd, gd, query, phid):
+async def get_github_commit_by_org(orgs,url,request_data,session,github_resp,rateLimitCount):
+    """
+    :Summary: make concurrent requests to get the users commits to wikimedia accounts on github two at a time.
+    :param url: URL to be fetched.
+    :param session: ClientSession to perform the API request.
+    :param request_data: data that is expected to be in the request but not in the url string link headers
+    :param github_resp: Global response array to which the response from the API has to be appended.
+    :return: None
+    """
+    createdStart = datetime.fromtimestamp(request_data["createdStart"])
+    createdEnd = datetime.fromtimestamp(request_data["createdEnd"])
+
+    dateRangeStart = createdStart
+    dateRangeEnd = createdEnd
+
+    """
+    <-----------------------------------------------------------
+    If time range between createdStart and createdEnd is greater than 6 months, divide it into two and fetch
+    each half seperately
+     """
+    if(dateRangeEnd - dateRangeStart) > timedelta(183):
+        dateRangeEnd = createdStart + timedelta(183)
+
+    loopCount = ceil((createdEnd - createdStart)/timedelta(183))
+    """-----------------------------------------------------/>"""
+
+
+    headers = {"Authorization":"token "+request_data["github_access_token"],
+               "Accept":"application/vnd.github.cloak-preview"}
+
+    orgs_filter = """user:{org_0}+user:{org_1}""".format(org_0=orgs[0],org_1=orgs[1])
+    query = """{url}+{orgs_filter}+committer-date:{dateRangeStartIsoFormat}..{dateRangeEndIsoFormat}"""
+
+
+    def getNextUrlOrNone(response):
+        url = None
+        link = response.headers.get("Link")
+        try:
+            for page in link.split(","):
+                if(page.endswith('rel="next"')):
+                    url = page.split(";")[0].split(">")[0].split("<")[1]
+                    break
+        except:
+            pass
+        return url
+
+    count = 0
+    while loopCount > 0:
+        count += 1
+        newURL = query.format(url=url,orgs_filter=orgs_filter,dateRangeStartIsoFormat=dateRangeStart.isoformat()+"Z",
+                              dateRangeEndIsoFormat=dateRangeEnd.isoformat()+"Z")
+        while newURL:
+            if rateLimitCount[0] == 30:
```

Comment:
Did it like this because the ratelimit in question only concerns github api and maybe it global might make it seem at first glance like it involves phab and gerrit APIs as well. But I can always make it global. Won't take more than 2 minutes